### PR TITLE
FEAT: Added categories on homepage and linked to filtered products

### LINKED ISSUE #8: Add categories on homepage
Create categories on home page and link them to product listing page showing products of only those categories.

### PR DESCRIPTION
**Pull request by shivanigangadharan**
Fixes #8 , #12 
Please only review changes from commit number [583a7c6](https://github.com/shivanigangadharan/ecom-app/pull/13/commits/583a7c6f702955497e75efe11789143bc7556a02), i.e., adding categories on Homepage. Rest of the commits are from previous branches which are in different PR. 

Files to review : homepage.js and homepage.css

### CODE REVIEW

File: src/pages/homepage/homepage.js [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+import { useFilter } from '../../context/filterContext';
+import axios from 'axios';
 
 export default function Homepage() {
+
+    const [categories, setCategories] = useState();
+
+    useEffect(async () => {
+        const res = await axios.get("/api/categories");
+        console.log(res);
```
Comment (junaidshaikh-js) [inline src/pages/homepage/homepage.js]: console logs should be removed before pushing your code

File: src/pages/homepage/homepage.js [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
     return (
         <div>
             <div className="homepage">
                 <div className="img-grid">
-                    <img alt="clothing" src="https://picsum.photos/id/1005/250/150" />
-                    <img alt="clothing" src="https://picsum.photos/id/1005/250/150" />
-                    <img alt="clothing" src="https://picsum.photos/id/1005/250/150" />
-                    <img alt="clothing" src="https://picsum.photos/id/1005/250/150" />
-                    <img alt="clothing" src="https://picsum.photos/id/1005/250/150" />
+                    <img className="category-img" onClick={() => handleCategory("womens clothing")} alt="clothing" src="https://images.unsplash.com/photo-1503342217505-b0a15ec3261c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80" />
```
Comment (junaidshaikh-js) [inline src/pages/homepage/homepage.js]: `alt` text can be more descriptive than 'clothing'  

I would suggest to give category name as the `alt` text

**Review by junaidshaikh-js — COMMENTED** [review]
Good work! 

Clean code. I suggested some points in the comments. 

Other than that everything looks fine. You can merge after doing the changes.

File: src/pages/homepage/homepage.js (hunk 1 above)
Comment (anshikaa12) [inline src/pages/homepage/homepage.js]: consoles to be removed!

**Review by anshikaa12 — APPROVED** [review]

File: src/pages/homepage/homepage.js (hunk 1 above)
Comment (shivanigangadharan) [reply src/pages/homepage/homepage.js]: Done

File: src/pages/homepage/homepage.js (hunk 2 above)
Comment (shivanigangadharan) [reply src/pages/homepage/homepage.js]: done this, thanks.

File: src/pages/homepage/homepage.js (hunk 1 above)
Comment (shivanigangadharan) [reply src/pages/homepage/homepage.js]: done

**Pull request merged**